sotodlib/preprocess/preprocess_util.py

```diff
@@ -871,301 +967,299 @@ def cleanup_obs(obs_id, policy_dir, errlog, configs, context=None,
+                    f"{out_dict_init['temp_file']}:{out_dict_init['db_data']['dataset']}")
+        proc_aman.save(out_dict_init['temp_file'], out_dict_init['db_data']['dataset'],
+                       overwrite)
         if save_archive:
-            cleanup_mandb(error, outputs_init, configs_init,
-                          logger=logger, overwrite=overwrite)
-
+            logger.info(f"Adding result to init db for {obs_id}: {group}")
+            cleanup_mandb(out_dict_init, (obs_id, group), (None, None, None),
+                          configs_init, logger=logger, overwrite=overwrite)
+        # Make init plots
+        if make_lmsi_init:
+            new_plots = os.path.join(configs_init["plot_dir"],
+                                 f'{str(aman.timestamps[0])[:5]}',
+                                 aman.obs_info.obs_id)
+            from pathlib import Path
+            import lmsi.core as lmsi
+
+            if os.path.exists(new_plots):
+                lmsi.core([Path(x.name) for x in Path(new_plots).glob("*.png")],
+                          Path(configs_init["lmsi_config"]),
+                          Path(os.path.join(new_plots, 'index.html')))
+
+        # Return if not running proc db
         if configs_proc is None:
-            return error, outputs_init, [obs_id, dets], aman
-        else:
-            try:
-                outputs_proc = save_group(obs_id, configs_proc, dets, context_proc, subdir='temp_proc')
-                init_fields = aman.preprocess._fields.copy()
-                init_fields.pop('valid_data', None)
-                logger.info(f"Generating new dependent preproc db entry for {obs_id} {dets}")
-                # pipeline for processing config
-                pipe_proc = Pipeline(configs_proc["process_pipe"], plot_dir=configs_proc["plot_dir"], logger=logger)
-                 # tags from context proc
-                tags_proc = np.array(context_proc.obsdb.get(aman.obs_info.obs_id, tags=True)['tags'])
-
-                if "tags" in aman._fields:
-                    aman.move("tags", None)
-                aman.wrap('tags', tags_proc)
+            logger.info(f"preproc_or_load_group finished successfully for {obs_id}:{group}")
+            return aman, out_dict_init, None, (None, None, None)
 
-                proc_aman, success = pipe_proc.run(aman)
+    # Run second layer
+    if (not db_proc_exist or overwrite) and configs_proc is not None:
+        try:
+            logger.info(f"Generating new proc db entry for {obs_id}: {group}")
+            init_fields = aman.preprocess._fields.copy()
+            init_fields.pop('valid_data', None)
+            tags_proc = np.array(context_proc.obsdb.get(aman.obs_info.obs_id,
+                                                        tags=True)['tags'])
+            if "tags" in aman._fields:
+                aman.move("tags", None)
+            aman.wrap('tags', tags_proc)
+
+            out_dict_proc = get_preproc_group_out_dict(obs_id, configs_proc,
```

Comment:
In get_groups you dropped context_init/context_proc input and set to configs_init['context_file'] I'd do that consistently everywhere including here.